### PR TITLE
fix(TagOverflow): use operational tag

### LINKED ISSUE #6002: Update TagOverflow to use  OperationalTag
Followup to #4358.

As previously discussed with @szinta @laurenmrice @guidari, we should update our `TagOverflow` and `TagSet` to use the new tag types.
```[tasklist]
### Tasks
- [ ] use operational tag for the `overflow tag` for `tag overflow`
- [ ] check if all tests are passing, fix any that fails
```

### PR DESCRIPTION
**Pull request by sangeethababu9223**
Closes #6002

Update TagOverflow to use OperationalTag

#### What did you change?
Replaced `Tag` with `OperationalTag`  for tag with click action.

#### How did you test and verify your work?
Storybook

### CODE REVIEW

**Review by devadula-nandan — APPROVED** [review]
LGTM

File: packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.tsx [hunk 1]
```diff
@@ -8,7 +8,14 @@
 import React, { useRef, forwardRef, Ref } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
-import { Link, Tag, Popover, PopoverContent } from '@carbon/react';
+import {
+  Link,
+  Tag,
+  Popover,
+  PopoverContent,
+  /**@ts-ignore */
```
Comment (davidmenendez) [inline packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.tsx:16]: you can alleviate the import error by including the import [here](https://github.com/carbon-design-system/ibm-products/blob/main/packages/ibm-products/src/custom-typings/index.d.ts)

**Review by davidmenendez — CHANGES_REQUESTED** [review]
one small note!

File: packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.tsx (hunk 1 above)
Comment (sangeethababu9223) [reply packages/ibm-products/src/components/TagOverflow/TagOverflowPopover.tsx]: Thanks @davidmenendez!
This is updated now.

**Review by devadula-nandan — APPROVED** [review]

**Review by davidmenendez — APPROVED** [review]

**Pull request merged**